src/app.py

```diff
@@ -173,13 +171,23 @@ def load_file(filename: str) -> dict:
         ["all", "public", "private", "internal"],
         key="repos_repo_type",
     )
+    # Date input for filtering repositories
+    col1, col2 = st.columns(2)
+    with col1:
+        start_date = st.date_input("Start Date", datetime.now().date() - pd.DateOffset(years=1), key="start_date_repo")
```

Comment:
I think it'd be a good idea to set the minimum date as the oldest date in the dataset. This means that when the user opens the dashboard, they can see the whole dataset.